src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -258,7 +264,16 @@ public static ResourceIdentifier of(String service, String instance, String type
         checkInstanceIsValid(instance);
         checkTypeIsValid(type);
         checkLocatorIsValid(locator);
-        return new ResourceIdentifier(service, instance, type, locator);
+
+        String safeInstance = instance == null ? "" : instance;
+        String resourceIdentifier =
+                RID_CLASS + SEPARATOR + service + SEPARATOR + safeInstance + SEPARATOR + type + SEPARATOR + locator;
```

Comment:
I suspect this will be more efficient if we bump the compile target from 1.8 to 11 per [jep280](https://openjdk.java.net/jeps/280).

Reply:
Interesting - I haven't really ever seen anyone use this spread out constructor (except in tests), so I would have normally avoided taking the bytecode bump just for this, but if you think it's tolerable now happy to put a PR in???
Presumably it would be a resourceidentifier 2.x?!?